### PR TITLE
fix: translate untranslated widgets

### PR DESCRIPTION
**Pull request by denyskon**
- Add missing strings to German locale
- translate markdown editor buttons, code widget, folder creation dialog buttons

### CODE REVIEW

File: packages/core/src/widgets/code/CodeControl.tsx [hunk 1]
```diff
@@ -51,7 +52,9 @@ function valueToOption(val: string | { name: string; label?: string }): {
   return { value: val.name, label: val.label || val.name };
 }
 
-const CodeControl: FC<WidgetControlProps<string | { [key: string]: string }, CodeField>> = ({
+const CodeControl: FC<
+  TranslatedProps<WidgetControlProps<string | { [key: string]: string }, CodeField>>
```
Comment (KaneFreeman) [inline packages/core/src/widgets/code/CodeControl.tsx:56]: `WidgetControlProps` already includes `t`. It is not necessary to wrap it in `TranslatedProps`.

**Comment by denyskon** [comment]
@KaneFreeman Do you have any idea how to fix this test? Did I do something wrong?

**Review by KaneFreeman — CHANGES_REQUESTED** [review]
The app will be switching away from passing `t` around (more than necessary) in `v4.0.0`. To help me out can you change to the new pattern of using the `useTranslate` hook instead of passing `t` down via `TranslatedProps`.

```ts
import useTranslate from '@staticcms/core/lib/hooks/useTranslate';

const t = useTranslate();
```

Example: https://github.com/StaticJsCMS/static-cms/blob/9876bd401f2313043054af2607d4ed11919a9980/packages/core/src/components/collections/entries/EntryListing.tsx#L54

**Comment by KaneFreeman** [comment]
> @KaneFreeman Do you have any idea how to fix this test? Did I do something wrong?

You made `t` a required prop for the Balloon toolbar, but did not update the test to pass it in. Switching to use the `useTranslate` hook will avoid this issue.

**Review by KaneFreeman — APPROVED** [review]

**Comment by KaneFreeman** [comment]
I will go ahead and fix the tests for this. Right now its based on the label/tooltip for querying those buttons, which is not translated and translations are mocked out in tests. I will need to make some adjustments to have them not rely on the translated strings.

**Comment by KaneFreeman** [comment]
Thanks for making these changes!

**Comment by denyskon** [comment]
Thanks, I really have not have enough overview to fix them....

And I'm happy to contribute as long as my skills allow me to :)

**Pull request merged**